File: js/src/maplestory/api/dto/character/characterVMatrixCoreEquipmentDto.ts
import { CharacterVMatrixCoreEquipmentDtoBody } from '@src/maplestory/api/response/character/characterVMatrixDtoBody';

/**
 * 캐릭터 V코어 정보
 */
class CharacterVMatrixCodeEquipmentDto {
  /**
   * 슬롯 인덱스
   */
  slotId: string;

  /**
   * 슬롯 레벨
   */
  slotLevel: number;

  /**
   * 코어 명
   */
  vCoreName: string | null;

  /**
   * 코어 타입
   */
  vCoreType: string | null;

  /**
   * 코어 레벨
   */
  vCoreLevel: number;

  /**
   * 코어에 해당하는 스킬 명
   */
  vCoreSkill1: string;

  /**
   * (강화 코어인 경우) 코어에 해당하는 두 번째 스킬 명
   */
  vCoreSkill2: string | null;

  /**
   * (강화 코어인 경우) 코어에 해당하는 세 번째 스킬 명
   */
  vCoreSkill3: string | null;

  constructor(obj: CharacterVMatrixCoreEquipmentDtoBody) {
    const {
      slot_id,
      slot_level,
      v_core_name,
      v_core_type,
      v_core_level,
      v_core_skill_1,
      v_core_skill_2,
      v_core_skill_3,
    } = obj;

    this.slotId = slot_id;
    this.slotLevel = slot_level;
    this.vCoreName = v_core_name;
    this.vCoreType = v_core_type;
    this.vCoreLevel = v_core_level;
    this.vCoreSkill1 = v_core_skill_1;
    this.vCoreSkill2 = v_core_skill_2;
    this.vCoreSkill3 = v_core_skill_3;
  }
}

export { CharacterVMatrixCodeEquipmentDto };

File: java/src/main/java/dev/spiralmoon/maplestory/api/tms/dto/character/CharacterVMatrixCoreEquipmentDTO.java
package dev.spiralmoon.maplestory.api.tms.dto.character;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.List;

/**
 * V 核心資訊
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class CharacterVMatrixCoreEquipmentDTO implements dev.spiralmoon.maplestory.api.common.dto.character.CharacterVMatrixCoreEquipmentDTO {

    /**
     * 欄位索引
     */
    @JsonProperty("slot_id")
    private String slotId;

    /**
     * 欄位等級
     */
    @JsonProperty("slot_level")
    private long slotLevel;

    /**
     * 核心名稱
     */
    @JsonProperty("v_core_name")
    private String vCoreName;

    /**
     * 核心類型
     */
    @JsonProperty("v_core_type")
    private String vCoreType;

    /**
     * 核心等級
     */
    @JsonProperty("v_core_level")
    private long vCoreLevel;

    /**
     * 對應核心的技能名稱
     */
    @JsonProperty("v_core_skill_1")
    private String vCoreSkill1;

    /**
     * (若為強化核心) 對應核心的第二個技能名稱
     */
    @JsonProperty("v_core_skill_2")
    private String vCoreSkill2;

    /**
     * (若為強化核心) 對應核心的第三個技能名稱
     */
    @JsonProperty("v_core_skill_3")
    private String vCoreSkill3;
}
